pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -395,6 +394,45 @@ func (cl *Client) addCRD(name string) {
 	cl.kinds[resourceGVK] = kc
 }
 
+// composedFilter offers a way to join multiple different object filters into a single one
+type composedFilter struct {
+	// The primary filter, which has a handler. Optional
+	filter kubetypes.DynamicObjectFilter
+	// Secondary filters (no handler allowed)
+	extra []func(obj any) bool
+}
+
+func (f composedFilter) Filter(obj any) bool {
+	for _, filter := range f.extra {
+		if !filter(obj) {
+			return false
+		}
+	}
+	if f.filter != nil {
+		return f.filter.Filter(obj)
```

Comment:
It seems the orders matter, and extras take first. I think we need to add more comments about this